cellprofiler/object.py

```diff
@@ -308,10 +309,22 @@ def relate_children(self, children):
         """
         if self.volumetric:
             histogram = self.histogram_from_labels(self.segmented, children.segmented)
+	    return self.relate_histogram(histogram)
         else:
-            histogram = self.histogram_from_ijv(self.ijv, children.ijv)
-
-        return self.relate_histogram(histogram)
+	    try:
+		child_hist=[]
+		parent_list=[]
+		parent_labels=self.segmented
+		parent_labels=numpy.where(parent_labels==0, numpy.NaN, parent_labels) #ignore the background
+		child_labels=children.segmented
+		for eachchild in range(1,numpy.max(child_labels)+1):
+		    child_hist.append(int(scipy.stats.mode((numpy.where(child_labels==eachchild, 1, numpy.NaN)) * parent_labels, axis=None, nan_policy='omit')[0][0]))
+		for eachparent in range(1,numpy.max(child_labels)+1):
+                    parent_list.append(child_hist.count(eachparent))
+		return numpy.array(child_hist), numpy.array(parent_list)		
+	    except:
```

Comment:
I'd highly recommend catching a specific exception here when you're able to determine if this is more efficient :)

Reply:
For sure!  Just waiting to see if a) it'll build and b) it'll run without dying on the dataset that currently dies when I try to run it.  I would try to add some list comprehensions, etc too. 

_A priori_ it seems logical to me that just checking what parent values are present in the child would be more memory efficient (if not necessarily clock-faster) than computing the whole coo matrix, but my instincts for this are 100% a WIP- do you have any sense on if you think this might work? I should know in the next couple hours I guess.

Reply:
I totally agree - I think that's a better heuristic, and then excluding those children from the coo matrix computation